app/screens/route-details/components/live-ride-sheet.tsx

```diff
@@ -51,24 +51,36 @@ export const LiveRideSheet = observer(function LiveRideSheet(props: { progress;
   )
 })
 
-const StopButton = (props: { loading: boolean } & PressableProps) => (
-  <Pressable
-    style={({ pressed }) => [
-      {
-        width: 42.5,
-        height: 42.5,
-        alignItems: "center",
-        justifyContent: "center",
-        backgroundColor: color.stop,
-        borderRadius: 30,
-      },
-    ]}
-    {...props}
-  >
-    {props.loading ? (
-      <ActivityIndicator color="white" />
-    ) : (
-      <Image source={require("../../../../assets/stop-rect.ios.png")} style={{ width: 17.5, height: 17.5 }} />
-    )}
-  </Pressable>
-)
+const STOP_BUTTON: ViewStyle = {
+  width: 42.5,
+  height: 42.5,
+  alignItems: "center",
+  justifyContent: "center",
+  backgroundColor: color.stop,
+  borderRadius: 30,
+}
+
+const StopButton = (props: { loading: boolean } & PressableProps) => {
+  const [isDisabled, setIsDisabled] = useState(true)
+
+  useEffect(() => {
+    // disable the button for 5 seconds
+    // this prevents frequent ride start & stop presses,
+    // and gives the request time to fullfill
+    const timeout = setTimeout(() => {
+      setIsDisabled(false)
+    }, 5000)
```

Comment:
Shouldn't we do that only when the request is loading? There's no need to disable the button for 5 seconds when the request is already fulfilled